File: src/main/java/com/uitnetwork/alexa/service/SpeechletService.java
package com.uitnetwork.alexa.service;

import static org.slf4j.LoggerFactory.getLogger;

import org.slf4j.Logger;
import org.springframework.beans.factory.annotation.Autowired;
import org.springframework.stereotype.Service;

import com.amazon.speech.json.SpeechletRequestEnvelope;
import com.amazon.speech.speechlet.*;
import com.uitnetwork.alexa.model.AlexaIntentRequest;

@Service
public class SpeechletService implements SpeechletV2 {
    private static final Logger logger = getLogger(TimeIntentService.class);

    @Autowired
    private IntroductionService introductionService;

    @Autowired
    private AlexaIntentRequestMappingService alexaIntentRequestMappingService;

    @Autowired
    private AlexaIntentRequestManager alexaIntentRequestManager;

    @Override
    public void onSessionStarted(SpeechletRequestEnvelope<SessionStartedRequest> requestEnvelope) {
        logger.info("onSessionStarted requestId={}, sessionId={}", requestEnvelope.getRequest().getRequestId(), requestEnvelope.getSession().getSessionId());
    }

    @Override
    public SpeechletResponse onLaunch(SpeechletRequestEnvelope<LaunchRequest> requestEnvelope) {
        logger.info("onLaunch requestId={}, sessionId={}", requestEnvelope.getRequest().getRequestId(), requestEnvelope.getSession().getSessionId());
        return introductionService.createIntrodutionSpeechletResponse(requestEnvelope);
    }

    @Override
    public SpeechletResponse onIntent(SpeechletRequestEnvelope<IntentRequest> requestEnvelope) {
        logger.info("onIntent requestId={}, sessionId={}", requestEnvelope.getRequest().getRequestId(), requestEnvelope.getSession().getSessionId());

        AlexaIntentRequest alexaIntentRequest = alexaIntentRequestMappingService.translate(requestEnvelope);
        return alexaIntentRequestManager.process(alexaIntentRequest);
    }

    @Override
    public void onSessionEnded(SpeechletRequestEnvelope<SessionEndedRequest> requestEnvelope) {
        logger.info("onSessionEnded requestId={}, sessionId={}", requestEnvelope.getRequest().getRequestId(), requestEnvelope.getSession().getSessionId());
    }
}

File: src/main/kotlin/com/uitnetwork/alexa/service/SpeechletService.kt
package com.uitnetwork.alexa.service

import com.amazon.speech.json.SpeechletRequestEnvelope
import com.amazon.speech.speechlet.*
import org.slf4j.LoggerFactory.getLogger
import org.springframework.beans.factory.annotation.Autowired
import org.springframework.stereotype.Service

@Service
class SpeechletService : SpeechletV2 {
    private val logger = getLogger(TimeIntentService::class.java)

    @Autowired
    private lateinit var introductionService: IntroductionService

    @Autowired
    private lateinit var alexaIntentRequestMappingService: AlexaIntentRequestMappingService

    @Autowired
    private lateinit var alexaIntentRequestManager: AlexaIntentRequestManager

    override fun onSessionStarted(requestEnvelope: SpeechletRequestEnvelope<SessionStartedRequest>) {
        logger.info("onSessionStarted requestId={}, sessionId={}", requestEnvelope.request.requestId, requestEnvelope.session.sessionId)
    }

    override fun onLaunch(requestEnvelope: SpeechletRequestEnvelope<LaunchRequest>): SpeechletResponse {
        logger.info("onLaunch requestId={}, sessionId={}", requestEnvelope.request.requestId, requestEnvelope.session.sessionId)
        return introductionService.createIntrodutionSpeechletResponse(requestEnvelope)
    }

    override fun onIntent(requestEnvelope: SpeechletRequestEnvelope<IntentRequest>): SpeechletResponse {
        logger.info("onIntent requestId={}, sessionId={}", requestEnvelope.request.requestId, requestEnvelope.session.sessionId)

        val alexaIntentRequest = alexaIntentRequestMappingService.translate(requestEnvelope)
        return alexaIntentRequestManager.process(alexaIntentRequest)
    }

    override fun onSessionEnded(requestEnvelope: SpeechletRequestEnvelope<SessionEndedRequest>) {
        logger.info("onSessionEnded requestId={}, sessionId={}", requestEnvelope.request.requestId, requestEnvelope.session.sessionId)
    }
}
